Extend `pybamm_install_odes` to include support for macOS systems (#3417)
pybamm-team/PyBaMM: docs/source/user_guide/installation/GNU-linux.rst

Optional - scikits.odes solver
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Users can install `scikits.odes <https://github.com/bmcage/odes>`__ to utilize the wrapped SUNDIALS ODE and DAE `solvers <https://docs.pybamm.org/en/latest/source/api/solvers/scikits_solvers.html>`__ in PyBaMM.

Reviewer: ```suggestion
Users can install `scikits.odes <https://github.com/bmcage/odes>`__ to utilize its interfaced SUNDIALS ODE and DAE `solvers <https://docs.pybamm.org/en/latest/source/api/solvers/scikits_solvers.html>`__ wrapped in PyBaMM.
```

A bit nitpicky; since the most of the work done by the solver is through the Python bindings that `scikits.odes` provides

   .. code:: bash

      apt install libopenblas-dev
      pybamm_install_odes

>    The ``pybamm_install_odes`` command, installed with PyBaMM, automatically downloads and installs the SUNDIALS library on your

Reviewer: ```suggestion
```

This line is present below as well, we might as well keep it in one place

Author: Ah yes, i'll also fix it's indentation to appear on both tabs.

   .. code:: bash

      brew install openblas
      pybamm_install_odes

   The ``pybamm_install_odes`` command is installed with PyBaMM. It automatically downloads and installs the SUNDIALS library on your

>    system (under ``~/.local``), before installing ``scikits.odes``. (Alternatively, one can install SUNDIALS without this script and run ``pip install pybamm[odes]`` to install ``pybamm`` with ``scikits.odes``.)

Reviewer: We should also instruct users on how to avoid failures while installing `scikits.odes` without this script. We were mentioning "Assuming that SUNDIALS was installed as described above", but if someone runs `pip install pybamm[odes]` now, it would still fail even if SUNDIALS is installed on the system because the SUNDIALS_INST environment variable would not be set. Above in the same file, we recommend running `brew install sundials` before `pip install pybamm`, so we can use the location Homebrew installs it to and ask users to set SUNDIALS_INST to that location. On an ARM-based machine for my case it is installed under `/opt/homebrew/Cellar/sundials/6.6.0`, but I think it would be under `usr/local/` on an Intel-based macOS machine. We could either check that and provide that ourselves or we could ask them to run `brew info sundials` to find the location for their case (which is how I myself found it). Note: this also shows that Homebrew and `pybamm_install_odes` install SUNDIALS under different locations (which is good because we don't want to interfere with Homebrew's files).

Perhaps adding a link to https://scikits-odes.readthedocs.io/en/stable/installation.html would also be helpful. Otherwise, we should _recommend_ the use of `pybamm_install_odes`. Please feel free to resolve this comment as necessary, though I will tag @Saransh-cpp for his input on this too.

   .. code:: bash

>       brew install openblas
>       pybamm_install_odes

Reviewer: Should we add `gcc` and `gfortran` here?

```suggestion
      brew install openblas gcc gfortran
      pybamm_install_odes
```

Author: Yes, better to add.

   The ``pybamm_install_odes`` command, installed with PyBaMM, automatically downloads and installs the SUNDIALS library on your
   system (under ``~/.local``), before installing ``scikits.odes``. (Alternatively, one can install SUNDIALS without this script and run ``pip install pybamm[odes]`` to install ``pybamm`` with ``scikits.odes``.)

To avoid installation failures when using ``pip install pybamm[odes]``, make sure to set the ``SUNDIALS_INST`` environment variable. If you have installed SUNDIALS using Homebrew, set the variable to the appropriate location. For example:

.. code:: bash

   export SUNDIALS_INST=$(brew --prefix sundials)

Ensure that the path matches the installation location on your system. You can verify the installation location by running:

.. code:: bash

   brew info sundials

Look for the installation path, and use that path to set the ``SUNDIALS_INST`` variable.

> Note: The location where Homebrew installs SUNDIALS might vary based on the system architecture (ARM or Intel). Adjust the path in the ``export SUNDIALS_INST`` command accordingly.

Reviewer: Can we link the `scikits.odes` installation docs here as well (because it requires other things like C compilers and `gfortran` that may not be present on someone's system)?

Author: Sure, will do that.

Reviewer: It might also make sense to rename this file to `gnu-linux-mac.rst` or something btw and fix any broken links, the heading at the top of the file says

```rst
GNU-Linux & MacOS
=================
```

but the file name is misleading for contributors

Author: Sure, thanks for pointing it out.
